.github/workflows/ci-complete.yml

```diff
@@ -41,8 +41,12 @@ permissions:
 jobs:
   upload-build-scan:
     # Skip this workflow if CI was run for anything other than "pull_request" (like "push").
-    # Also skip this workflow if the PR was from apache/kafka. Those will have already published the build scan in CI.
-    if: ${{ github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.head_repository.full_name != 'apache/kafka' }}
+    # Skip this workflow if the PR was from apache/kafka. Those will have already published the build scan in CI.
+    # Skip this workflow if the run was skipped or cancelled
+    if: |
+      github.event.workflow_run.event == 'pull_request' && 
+      github.event.workflow_run.head_repository.full_name != 'apache/kafka' &&
+       (github.event.workflow_run.conclusion == 'success' || github.event.workflow_run.conclusion == 'failure')
```

Comment:
With this filter, the build scan must be existent, right? if so, can task "Handle missing scan" leverage the "gh-api-update-status" to produce failure status as the "missing" is illegal?

Reply:
> With this filter, the build scan must be existent, right? 

Except in cases where gradle exited abnormally (i.e., no build run). I guess this could happen in a PR that has a syntax error in build.gradle. In this case, the CI workflows will all be failed and this workflow would fail too.

We could use "Handle missing scan" to set an error status on the "Gradle Build Scan" checks. Seems reasonable to me